js/plugins/google-genai/src/vertexai/gemini.ts

```diff
@@ -537,6 +560,17 @@ export function defineModel(
         };
       }
 
+      if (retrievalConfig) {
+        if (!toolConfig) {
+          toolConfig = {};
+        }
+        toolConfig.retrievalConfig = {};
+        if (retrievalConfig.latLng) {
+          toolConfig.retrievalConfig.latLng = { ...retrievalConfig.latLng };
+        }
+        toolConfig.retrievalConfig.languageCode = retrievalConfig.languageCode;
```

Comment:
how about something like this, with passthrough built-in?

```suggestion
        toolConfig.retrievalConfig = {...retrievalConfig};
```

Reply:
yes, but with copy not ref.